examples-site/react-components/example-client.tsx

```diff
@@ -0,0 +1,31 @@
+import { ElementType, StrictMode } from 'react'
+import { hydrateRoot } from 'react-dom/client'
+
+function getComponent(callback: (value: ElementType) => void) {
+  const reExp = /^\/react-components\/([\w|-]+)\/([\w|-]+)(\/$|$)/
+
+  const match = window.location.pathname.match(reExp)
+  const ErrorMessage = () => <div>Error finding component</div>
+  if (match) {
+    const [component, exampleName] = match.slice(1, 3)
+
+    import(
+      /* webpackInclude: /\/\w+\/__examples__\/(\w|-)+\.tsx$/ */
+      `../../src/react/${component}/__examples__/${exampleName}`
+    ).then((module) => callback(module?.default || ErrorMessage))
+    return
+  }
+  callback(ErrorMessage)
+}
+
+getComponent((Component) => {
```

Comment:
I find use of a callback makes things harder to follow. What about using an async function that is just called, or an IIFE? For example (very rough code):

```
function hydrate(Component) {
  ...
}

(async () => {
  try {
    const { default: component } = await import(...)
    hydrate(component)
  } catch (err) {
    hydrate(err.message)
  }
})();
```